### DISCUSSION TITLE
Debugging triple git commit runs over tramp

### DISCUSSION THREAD

**Discussion by kohnish** [magit/magit#5056] [Support]
It seems every time I make a git commit with magit over tramp, I get the following message.

```
Enabled verbose refreshing
Running git commit --

Running git commit --

Running git commit --
Type C-c C-c to finish, C-c C-k to cancel, and M-p and M-n to recover older messages
```
  
It's probably a bug from tramp but I'm looking for some workaround to stop this from happening and hopefully speed up and make it more stable, because I believe at this point rather than trying to fix tramp, making plugins tramp-friendly is faster and more future proof approach.

**Comment by kohnish** [comment]
Magit from melpa gets me, so maybe it's a magit bug. It's still strangely slow.
```
Running git commit --
Only whitespace and/or comments; message not saved

Only whitespace and/or comments; message not saved

Only whitespace and/or comments; message not saved
```

**Comment by tarsius** [comment]
I debug Tramp issues in bursts (because I don't use it myself and when I have to use it to debug its interaction with Magit, I always have to relearn how to hold it correctly so that it works at all). The next round will probably be early next year.

**Reply by tarsius** [reply]
Let me know if this still happens and I will reopen.
